src/fsharp/FSharp.Core/async.fs

```diff
@@ -198,6 +198,7 @@ namespace Microsoft.FSharp.Control
             else
                 // NOTE: this must be a tailcall
                 cont res
+        // TODO: Implement System.Threading.IThreadPoolWorkItem when we can.
```

Comment:
Remove this TODO and add a separate issue about this?  I actually didn't know about this interface and am not aware of what implementing it would do.

Reply:
Done. This interface allows scheduling work to the thread pool from an object without extra allocations (we are allocating _3_ delegates per trampoline holder which might not all be used).

I will look into it again when `FSharp.Core` targets .NET 5. Until then, I find no need to create an issue.